components/style/properties/longhand/effects.mako.rs

```diff
@@ -344,31 +344,43 @@ ${helpers.predefined_type("opacity",
         use std::ascii::AsciiExt;
         use values::specified::Length;
 
+        fn parse_argument(input: &mut Parser) -> Result<Option<Length>, ()> {
+            if input.try(|input| input.expect_ident_matching("auto")).is_ok() {
+                Ok(None)
+            } else {
+                Length::parse(input).map(Some)
+            }
+        }
+
         if input.try(|input| input.expect_ident_matching("auto")).is_ok() {
             return Ok(SpecifiedValue(None))
         }
         if !try!(input.expect_function()).eq_ignore_ascii_case("rect") {
             return Err(())
         }
-        let sides = try!(input.parse_nested_block(|input| {
```

Comment:
`parse_nested_block` is still needed. Without it, you’re ignoring the stuff between `rect(` and `)`, and then parsing four arguments that come *after* the function.

To fix this, move the rest of this function (from `let top = …;` to `Ok(…)`) into a 
```rust
try!(input.parse_nested_block(|input| {
    // …
}))`
```

block.